tests/Unit/Domain/Creators/Test_RotatedIntervals.cpp

```diff
@@ -104,35 +115,154 @@ void test_rotated_intervals() {
+          expected_boundary_conditions[0][Direction<1>::lower_xi()]
+              ->get_clone(),
+          periodic->get_clone(), Options::Context{false, {}, 1, 1}),
+      Catch::Matchers::Contains("Both the upper and lower boundary condition "
+                                "must be set to periodic if"));
+  CHECK_THROWS_WITH(
+      creators::RotatedIntervals(
+          lower_bound, midpoint, upper_bound, refinement_level[0],
+          {{{{grid_points[0][0], grid_points[1][0]}}}}, periodic->get_clone(),
+          expected_boundary_conditions[0][Direction<1>::lower_xi()]
+              ->get_clone(),
+          Options::Context{false, {}, 1, 1}),
+      Catch::Matchers::Contains("Both the upper and lower boundary condition "
+                                "must be set to periodic if"));
+}
+
+void test_rotated_intervals_factory() {
+  const OrientationMap<1> flipped{
+      std::array<Direction<1>, 1>{{Direction<1>::lower_xi()}}};
+  {
+    INFO("Rotated intervals factory, no boundary condition");
+    const auto domain_creator = TestHelpers::test_factory_creation<
+        DomainCreator<1>, domain::OptionTags::DomainCreator<1>,
+        TestHelpers::domain::BoundaryConditions::
+            MetavariablesWithoutBoundaryConditions<1>>(
+        "RotatedIntervals:\n"
+        "  LowerBound: [0.0]\n"
+        "  Midpoint:   [0.5]\n"
+        "  UpperBound: [1.0]\n"
+        "  IsPeriodicIn: [True]\n"
+        "  InitialGridPoints: [[3,2]]\n"
+        "  InitialRefinement: [2]\n");
+    const auto* rotated_intervals_creator =
+        dynamic_cast<const creators::RotatedIntervals*>(domain_creator.get());
+    test_rotated_intervals_construction(
+        *rotated_intervals_creator, {{0.0}}, {{0.5}}, {{1.0}}, {{{3}}, {{2}}},
+        {{{2}}, {{2}}},
+        std::vector<DirectionMap<1, BlockNeighbor<1>>>{
+            {{Direction<1>::lower_xi(), {1, flipped}},
+             {Direction<1>::upper_xi(), {1, flipped}}},
+            {{Direction<1>::lower_xi(), {0, flipped}},
+             {Direction<1>::upper_xi(), {0, flipped}}}},
+        std::vector<std::unordered_set<Direction<1>>>{{}, {}}, {});
+  }
+  {
+    INFO("Rotated intervals factory, with boundary condition");
+    const auto domain_creator = TestHelpers::test_factory_creation<
+        DomainCreator<1>, domain::OptionTags::DomainCreator<1>,
+        TestHelpers::domain::BoundaryConditions::
+            MetavariablesWithBoundaryConditions<1>>(
+        "RotatedIntervals:\n"
+        "  LowerBound: [0.0]\n"
+        "  Midpoint:   [0.5]\n"
+        "  UpperBound: [1.0]\n"
+        "  InitialGridPoints: [[3,2]]\n"
+        "  InitialRefinement: [2]\n"
+        "  BoundaryConditions:\n"
+        "    LowerBoundary:\n"
+        "      TestBoundaryCondition:\n"
+        "        Direction: lower-xi\n"
+        "        BlockId: 0\n"
+        "    UpperBoundary:\n"
+        "      TestBoundaryCondition:\n"
+        "        Direction: upper-xi\n"
```

Comment:
So I understand, why is this not `lower-xi` since the block is flipped?

Reply:
They are just arbitrary options to the boundary condition class used to distinguish the boundary conditions so that, e.g. the upper and lower boundary conditions don't compare equal

Reply:
Ok, perhaps add a comment?